pceplib/pcep_msg_objects_encoding.c

```diff
@@ -0,0 +1,1483 @@
+    obj_body_buf[0] = nopath->ni;
+    obj_body_buf[1] = ((nopath->flag_c == true) ? OBJECT_NOPATH_FLAG_C : 0x00);
+
+    return LENGTH_1WORD;
+}
+
+uint16_t pcep_encode_obj_association(struct pcep_object_header *hdr, struct pcep_versioning *versioning, uint8_t *obj_body_buf)
+{
+    (void) versioning;
+    uint16_t *uint16_ptr = (uint16_t *) obj_body_buf;
+    uint32_t *uint32_ptr = (uint32_t *) obj_body_buf;
+    if (hdr->object_type == PCEP_OBJ_TYPE_ASSOCIATION_IPV4)
+    {
+        struct pcep_object_association_ipv4 *ipv4 = (struct pcep_object_association_ipv4 *) hdr;
+        obj_body_buf[3] = (ipv4->R_flag == true ? OBJECT_ASSOCIATION_FLAG_R : 0x00);
+        uint16_ptr[2] = htons(ipv4->association_type);
+        uint16_ptr[3] = htons(ipv4->association_id);
+        uint32_ptr[2] = ipv4->src.s_addr;
+
+        return LENGTH_3WORDS;
+    }
+    else
+    {
+        struct pcep_object_association_ipv6 *ipv6 = (struct pcep_object_association_ipv6 *) hdr;
+        obj_body_buf[3] = (ipv6->R_flag == true ? OBJECT_ASSOCIATION_FLAG_R : 0x00);
+        uint16_ptr[2] = htons(ipv6->association_type);
+        uint16_ptr[3] = htons(ipv6->association_id);
+        uint32_ptr[2] = ipv6->src.ipv6_u.__u6_addr32[0];
+        uint32_ptr[3] = ipv6->src.ipv6_u.__u6_addr32[1];
+        uint32_ptr[4] = ipv6->src.ipv6_u.__u6_addr32[2];
+        uint32_ptr[5] = ipv6->src.ipv6_u.__u6_addr32[3];
+
+        return LENGTH_6WORDS;
+    }
+}
+
+uint16_t pcep_encode_obj_endpoints(struct pcep_object_header *hdr, struct pcep_versioning *versioning, uint8_t *obj_body_buf)
+{
+    (void) versioning;
+    uint32_t *uint32_ptr = (uint32_t *) obj_body_buf;
+    if (hdr->object_type == PCEP_OBJ_TYPE_ENDPOINT_IPV4)
+    {
+        struct pcep_object_endpoints_ipv4 *ipv4 = (struct pcep_object_endpoints_ipv4 *) hdr;
+        uint32_ptr[0] = ipv4->src_ipv4.s_addr;
+        uint32_ptr[1] = ipv4->dst_ipv4.s_addr;
+
+        return LENGTH_2WORDS;
+    }
+    else
+    {
+        struct pcep_object_endpoints_ipv6 *ipv6 = (struct pcep_object_endpoints_ipv6 *) hdr;
+        uint32_ptr[0] = ipv6->src_ipv6.ipv6_u.__u6_addr32[0];
+        uint32_ptr[1] = ipv6->src_ipv6.ipv6_u.__u6_addr32[1];
+        uint32_ptr[2] = ipv6->src_ipv6.ipv6_u.__u6_addr32[2];
+        uint32_ptr[3] = ipv6->src_ipv6.ipv6_u.__u6_addr32[3];
+
+        uint32_ptr[4] = ipv6->dst_ipv6.ipv6_u.__u6_addr32[0];
+        uint32_ptr[5] = ipv6->dst_ipv6.ipv6_u.__u6_addr32[1];
+        uint32_ptr[6] = ipv6->dst_ipv6.ipv6_u.__u6_addr32[2];
+        uint32_ptr[7] = ipv6->dst_ipv6.ipv6_u.__u6_addr32[3];
+
+        return LENGTH_8WORDS;
+    }
+}
+
+uint16_t pcep_encode_obj_bandwidth(struct pcep_object_header *hdr, struct pcep_versioning *versioning, uint8_t *obj_body_buf)
+{
+    (void) versioning;
+    struct pcep_object_bandwidth *bandwidth = (struct pcep_object_bandwidth *) hdr;
+    uint32_t *uint32_ptr = (uint32_t *) obj_body_buf;
+    /* Seems like the compiler doesnt correctly copy the float, so memcpy() it */
```

Comment:
I'm curious what was here before / what this comment means

Reply:
This was the change